MeshState[] states
float64[] yaws
float64[] arc_lengths